### PR TITLE
Fill out topology counting tutorial

### LINKED ISSUE #93: Tutorial: combinatorics
We need a tutorial to describe how to use features such as the [topology counter](https://tskit.dev/tskit/docs/stable/python-api.html#tskit.TopologyCounter). See e.g. the last comment at https://github.com/tskit-dev/tskit/issues/1462

### PR DESCRIPTION
**Pull request by hyanwong**
This has been on the to-do for while, and would have been useful to @duncanMR and others. I managed to get it written over the weekend. Fixes #93

### CODE REVIEW

File: counting_topologies.md [hunk 1]
```diff
@@ -16,13 +16,309 @@ kernelspec:
...
+
+The number of possible topologies for a tree with $N$ tips
+grows very rapidly with $N$. For instance, with 10 tips, there are
+282,137,824 possible topologies.
+
+For this reason, the {func}`~tskit.all_trees`, {func}`~tskit.all_tree_shapes` and
+{func}`~tskit.all_tree_labellings` methods do not return a list of trees
+but an iterator over the trees. This means it is perfectly possible to start
+iterating over (say) all tree shapes for a tree of 100 leaves, but
+the iterator will not finish before the death of our galaxy
```
Comment (benjeffery) [inline counting_topologies.md:191]:
```suggestion
the iterator will not finish before the death of our galaxy.
```

**Review by benjeffery — APPROVED** [review]
Great, I built and checked the HTML and looks good - only found one minor typo.

**Comment by hyanwong** [comment]
Thanks Ben - well spotted. I put the galaxy thing in for you!

Merge whenever you like.

**Comment by benjeffery** [comment]
> Thanks Ben - well spotted. I put the galaxy thing in for you!
> 
> Merge whenever you like.

Heh! I thought it was cool!

**Pull request merged**